Calculator.Tests/FunctionTests.cs

```diff
@@ -1,24 +1,110 @@
     [Test]
-    public void AddReturnsCorrectAnswer()
+    [TestCase(5, 4, 9)]    // Big positive number first, Small positive number second
+    [TestCase(4, 6, 10)]  //Small positive number, Big positive Number
+    [TestCase(-10, 6, -4)] //Big Negative Number, Small number
+    [TestCase(-12, 24, 12)] //Small Negative Number, Big Positive Number
+    [TestCase(-13, -7, -20)] //Big Negative Number, Small Negative Number
+    public void AddReturnsCorrectAnswer(int a, int b, int expectedSum)
     {
-        // arrange
+        // Arrange done by mapping TestCase Variables to Method Parameters
+        /*
         int a = 1;
         int b = 2;
         int expectedSum = 3;
+        */
 
         // act
-        int actualSum = Add(a, b);
+        var actualSum = Add(a, b);
 
         // assert
         Assert.AreEqual(expectedSum, actualSum);
     }
+
+    /*Fuction Name: SubtractReturnsCorrectAnswer
+    **Parameters: N/A
+    **Return: void
+    */
+    [Test]
+    [TestCase(5, 4, 1)]      // Big positive number first, Small positive number second
+    [TestCase(4, 6, -2)]     //Small positive number, Big positive Number
+    [TestCase(-10, 6, -16)]  //Big Negative Number, Small number
+    [TestCase(-12, 24, -36)] //Small Negative Number, Big Positive Number
+    [TestCase(-13, -7, -6)]  //Big Negative Number, Small Negative Number
+    [TestCase(-4, -12, 8)]   //Small Negative Number, Big Negative Number
+    public void SubtractReturnsCorrectAnswer(int a, int b, int expectedDiff)
+    {
+        // Arrange done by mapping TestCase Variables to Method Parameters
+        /*
+        int a = TestCaseAttribute(a);
+        int b;
+        int expectedDiff = -1;
+        */
+
+        // act
+        var actualDiff = Subtract(a, b);
+
+        // assert
+        Assert.AreEqual(expectedDiff, actualDiff);
+    }
+
+    /*Fuction Name: MultiplyReturnsCorrectAnswer
+    **Parameters: N/A
+    **Return: void
+    */
+    [Test]
+    [TestCase(5, 4, 20)]    //Positive, Positive
+    [TestCase(3, -4, -12)]  //Negative, Negative
+    [TestCase(-10, -5, 50)] //Negative, Negative
+
+    public void MultiplyReturnsCorrectAnswer(int a, int b, int expectedProduct)
+    {
+        // Arrange done by mapping TestCase Variables to Method Parameters
+        /*
+        int a;
+        int b;
+        int expectedProduct = 2;
+        */
+
+        // act
+        var actualProduct = Multiply(a, b);
+
+        // assert
+        Assert.AreEqual(expectedProduct, actualProduct);
+    }
+
+    /*Fuction Name: DivideReturnsCorrectAnswer
+    **Parameters: N/A
+    **Return: void
+    */
+    [Test]
+    [TestCase(10, 2, 5)]    // Big positive Numerator, Small positive Denominator
+    [TestCase(3, 4, 0.75)]  //Small Positive Numerator, Big Positive Denominator
+    [TestCase(-14, 7, -2)] //Big Negative Numerator, Small Positive Denominator
+    [TestCase(-12, 36, -0.33333333333333331d)] //Small Negative Numerator, Big Positive Number
```

Comment:
This case is interesting. This will relate to another comment below that I'll make regarding tests that compare two doubles to account for IEEE 754 precision problems. When you implement what I suggest below you can change the expected value for this test case to go to the precision you set for the Assert tolerance.